usr/lib/common/pkcs_utils.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * COPYRIGHT (c) International Business Machines Corp. 2020
+ *
+ * This program is provided under the terms of the Common Public License,
+ * version 1.0 (CPL-1.0). Any use, reproduction or distribution for this
+ * software constitutes recipient's acceptance of CPL-1.0 terms which can be
+ * found in the file LICENSE file or at
+ * https://opensource.org/licenses/cpl1.0.php
+ */
+
+#ifndef PKCS_UTILS_H
+#define PKCS_UTILS_H
+
+#include "pkcs11types.h"
+
+#define MASTER_KEY_SIZE       64
+
+#define HASH_SHA1   1
+#define HASH_MD5    2
+
+#define compute_sha1(a,b,c)     compute_hash(HASH_SHA1,b,a,c)
+#define compute_md5(a,b,c)      compute_hash(HASH_MD5,b,a,c)
+
+int compute_hash(int hash_type, int buf_size, char *buf, char *digest);
+
+CK_RV local_rng(CK_BYTE *output, CK_ULONG bytes);
+
+CK_RV aes_256_wrap(unsigned char out[40], const unsigned char in[32],
+                   const unsigned char kek[32]);
+
+CK_RV aes_256_unwrap(unsigned char key[32], const unsigned char in[40],
+                     const unsigned char kek[32]);
+
+CK_RV aes_256_gcm_seal(unsigned char *out, unsigned char tag[16],
+                       const unsigned char *aad, size_t aadlen,
+                       const unsigned char *in, size_t inlen,
+                       const unsigned char key[32],
+                       const unsigned char iv[12]);
+
+CK_RV get_pin(char **pin, size_t *pinlen);
+
+CK_RV verify_pins(char *data_store, char *sopin, unsigned long sopinlen,
+                  char *userpin, unsigned long userpinlen);
+
+/* Log levels */
```

Comment:
Those are also duplicates from common code. If we ever share this file between tools and OCK base code, then we should have it only here, and remove it from OCK common code.

Reply:
Yes, this is the intention. We will have to do some cleanup here.